src/slice.rs

```diff
@@ -157,7 +157,8 @@ impl<'a, T> NonEmptyIterator for NEChunks<'a, T> {
         } else if self.index >= self.tail.len() {
             None
         } else {
-            let end = self.index + self.window.get();
+            // Ensure we never go out of bounds
+            let end = min(self.index + self.window.get(), self.tail.len());
```

Comment:
Number types already have a `min` method provided by the `Ord` trait, so you don't need to important an extra function just for that.